dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -7433,3 +7706,84 @@ fn datetime_opt_rfc3339_concise(t: &Option<DateTime<Utc>>) -> String {
     t.map(|t| t.to_rfc3339_opts(chrono::format::SecondsFormat::Millis, true))
         .unwrap_or_else(|| "-".to_string())
 }
+
+async fn cmd_db_zpool_list(
+    opctx: &OpContext,
+    datastore: &DataStore,
+) -> Result<(), anyhow::Error> {
+    let zpools = datastore.zpool_list_all_external_batched(opctx).await?;
+
+    let Some(latest_collection) =
+        datastore.inventory_get_latest_collection(opctx).await?
+    else {
+        bail!("no latest inventory found!");
+    };
+
+    let mut zpool_total_size: HashMap<Uuid, i64> = HashMap::new();
+
+    for (_, sled_agent) in latest_collection.sled_agents {
+        for zpool in sled_agent.zpools {
+            zpool_total_size
+                .insert(zpool.id.into_untyped_uuid(), zpool.total_size.into());
+        }
+    }
+
+    #[derive(Tabled)]
+    #[tabled(rename_all = "SCREAMING_SNAKE_CASE")]
+    struct ZpoolRow {
+        id: Uuid,
+        time_deleted: String,
+        sled_id: Uuid,
+        physical_disk_id: Uuid,
+        total_size: i64,
+        control_plane_storage_buffer: i64,
+    }
+
+    let rows: Vec<_> = zpools
+        .into_iter()
+        .map(|(p, _)| {
+            let zpool_id = p.id().into_untyped_uuid();
+            ZpoolRow {
+                id: zpool_id,
+                time_deleted: match p.time_deleted() {
+                    Some(t) => t.to_string(),
+                    None => String::from(""),
+                },
+                sled_id: p.sled_id,
+                physical_disk_id: p.physical_disk_id.into_untyped_uuid(),
+                total_size: *zpool_total_size.get(&zpool_id).unwrap(),
+                control_plane_storage_buffer: p.control_plane_storage_buffer(),
+            }
+        })
+        .collect();
+
+    let table = tabled::Table::new(rows)
+        .with(tabled::settings::Style::psql())
+        .with(tabled::settings::Padding::new(0, 1, 0, 0))
+        .to_string();
+
+    println!("{}", table);
+
+    Ok(())
+}
+
+async fn cmd_db_zpool_set_storage_buffer(
```

Comment:
I know there's a `--destructive` flag for db ops, should that gate this operation?

Reply:
definitely, that's required by accepting the (unused) `DestructiveOperationToken` :)